invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/imageDeleted.ts

```diff
@@ -1,57 +1,72 @@
 import { logger } from 'app/logging/logger';
 import { resetCanvas } from 'features/canvas/store/canvasSlice';
 import { controlNetReset } from 'features/controlNet/store/controlNetSlice';
+import { imageDeletionConfirmed } from 'features/deleteImageModal/store/actions';
+import { isModalOpenChanged } from 'features/deleteImageModal/store/slice';
 import { selectListImagesBaseQueryArgs } from 'features/gallery/store/gallerySelectors';
 import { imageSelected } from 'features/gallery/store/gallerySlice';
-import { imageDeletionConfirmed } from 'features/imageDeletion/store/actions';
-import { isModalOpenChanged } from 'features/imageDeletion/store/imageDeletionSlice';
 import { nodeEditorReset } from 'features/nodes/store/nodesSlice';
 import { clearInitialImage } from 'features/parameters/store/generationSlice';
 import { clamp } from 'lodash-es';
 import { api } from 'services/api';
 import { imagesApi } from 'services/api/endpoints/images';
+import { imagesAdapter } from 'services/api/util';
 import { startAppListening } from '..';
 
-/**
- * Called when the user requests an image deletion
- */
-export const addRequestedImageDeletionListener = () => {
+export const addRequestedSingleImageDeletionListener = () => {
   startAppListening({
     actionCreator: imageDeletionConfirmed,
     effect: async (action, { dispatch, getState, condition }) => {
-      const { imageDTO, imageUsage } = action.payload;
+      const { imageDTOs, imagesUsage } = action.payload;
 
-      dispatch(isModalOpenChanged(false));
+      if (imageDTOs.length !== 1 || imagesUsage.length !== 1) {
+        // handle multiples in separate listener
```

Comment:
Any reason single and multi image deletion couldn't be the same listener and backend endpoint? Like, single image could just send a list of images of size 1. Just seems like this could be annoying to maintain(or forgotten about) in the future.

Reply:
I had approached the same conclusion but am a bit burnt out on the gallery stuff and would prefer to leave it for a future cleanup

Reply:
Raised in  #4142